src/AST/Class.cs

```diff
@@ -102,6 +102,9 @@ public class Class : DeclarationContext
         // True if the class represents a static class.
         public bool IsStatic;
 
+        // True if declared with 'struct' keyword
+        public bool DeclaredStruct;
```

Comment:
I wouldn't say so. Basically, C# structs are limited compared to C++ structs so this code simply means that we apply these limitations on top if the type is a (C++) struct.